sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/StoreClientFactory.java

```diff
@@ -13,28 +13,36 @@
 //  https://msdata.visualstudio.com/CosmosDB/SDK/_workitems/edit/262496
 
 public class StoreClientFactory implements AutoCloseable {
+
     private final Configs configs;
     private final int maxConcurrentConnectionOpenRequests;
     private final int requestTimeoutInSeconds;
     private final Protocol protocol;
     private final TransportClient transportClient;
+    private final boolean enableTransportClientSharing;
     private volatile boolean isClosed;
 
     public StoreClientFactory(
         Configs configs,
         int requestTimeoutInSeconds,
         int maxConcurrentConnectionOpenRequests,
-        UserAgentContainer userAgent) {
+        UserAgentContainer userAgent,
+        boolean enableTransportClientSharing) {
 
         this.configs = configs;
         this.protocol = configs.getProtocol();
         this.requestTimeoutInSeconds = requestTimeoutInSeconds;
         this.maxConcurrentConnectionOpenRequests = maxConcurrentConnectionOpenRequests;
+        this. enableTransportClientSharing = enableTransportClientSharing;
 
         if (protocol == Protocol.HTTPS) {
             this.transportClient = new HttpTransportClient(configs, requestTimeoutInSeconds, userAgent);
         } else if (protocol == Protocol.TCP){
-            this.transportClient = new RntbdTransportClient(configs, requestTimeoutInSeconds, userAgent);
+            if (enableTransportClientSharing) {
```

Comment:
Can the sharing be transport agnostic?

Reply:
I thought about that earlier. However port-reuse option (coming later) is only applicable to TCP. 

However it is worth to add it for Direct Https regardless of port-reuse. I will make the change to make it agnostic. sure

Reply:
made it transport agnostic. addressed.